packages/@aws-cdk/aws-appconfig-alpha/lib/deployment-strategy.ts

```diff
@@ -0,0 +1,284 @@
+    this.growthFactor = props.rolloutStrategy.growthFactor;
+    this.description = props.description;
+    this.finalBakeTimeInMinutes = props.rolloutStrategy.finalBakeTime?.toMinutes();
+    this.growthType = props.rolloutStrategy.growthType;
+    this.name = props.name || Names.uniqueResourceName(this, {
+      maxLength: 64,
+      separator: '-',
+    });
+
+    const resource = new CfnDeploymentStrategy(this, 'Resource', {
+      name: this.name,
+      deploymentDurationInMinutes: this.deploymentDurationInMinutes,
+      growthFactor: this.growthFactor,
+      replicateTo: 'NONE',
+      description: this.description,
+      finalBakeTimeInMinutes: this.finalBakeTimeInMinutes,
+      growthType: this.growthType,
+    });
+    this._cfnDeploymentStrategy = resource;
+
+    this.deploymentStrategyId = this._cfnDeploymentStrategy.ref;
+    this.deploymentStrategyArn = this.stack.formatArn({
+      service: 'appconfig',
+      resource: 'deploymentstrategy',
+      resourceName: this.deploymentStrategyId,
+    });
+  }
+}
+
+/**
+ * Defines the growth type of the deployment strategy.
+ */
+export enum GrowthType {
+  LINEAR = 'LINEAR',
+  EXPONENTIAL = 'EXPONENTIAL',
+}
+
+/**
+ * Defines the deployment strategy ID's of AWS AppConfig predefined strategies.
+ */
+export enum PredefinedDeploymentStrategyId {
+  CANARY_10_PERCENT_20_MINUTES = 'AppConfig.Canary10Percent20Minutes',
+  LINEAR_50_PERCENT_EVERY_30_SECONDS = 'AppConfig.Linear50PercentEvery30Seconds',
+  LINEAR_20_PERCENT_EVERY_6_MINUTES = 'AppConfig.Linear20PercentEvery6Minutes',
+  ALL_AT_ONCE = 'AppConfig.AllAtOnce',
+}
+
+/**
+ * Defines the rollout strategy for a deployment strategy and includes the growth factor,
+ * deployment duration, and growth type. The growth factor defines the percentage of targets
+ * to receive a deployed configuration during each interval. The deployment duration defines
+ * the total amount of time for a deployment to last. The growth type defines the algorithm
+ * used to define how percentage grows over time. The growth type can be modeled as either
+ * GrowthType.LINEAR or GrowthType.EXPONENTIAL.
+ */
+export abstract class RolloutStrategy {
+  public static readonly CANARY_10_PERCENT_20_MINUTES = RolloutStrategy.exponential(10, Duration.minutes(20), Duration.minutes(10));
+  public static readonly LINEAR_50_PERCENT_EVERY_30_SECONDS = RolloutStrategy.linear(50, Duration.minutes(1), Duration.minutes(1));
+  public static readonly LINEAR_20_PERCENT_EVERY_6_MINUTES = RolloutStrategy.linear(20, Duration.minutes(30), Duration.minutes(30));
+  public static readonly ALL_AT_ONCE = RolloutStrategy.linear(100, Duration.minutes(0), Duration.minutes(10));
+
+  /**
+   * @returns A linear rollout strategy.
+   *
+   * @param growthFactor The growth factor of the strategy
+   * @param deploymentDuration The duration of the deployment
+   */
+  public static linear(growthFactor: number, deploymentDuration: Duration, finalBakeTime?: Duration): RolloutStrategy {
```

Comment:
One final thing: please change this signature to receive a data structure, instead of three separate parameters. This is especially important in this case, where we have two arguments of the same type, which tends to confuse users.